package/src/ssr/NextSSRApolloClient.tsx

```diff
@@ -0,0 +1,169 @@
+import {
+  ApolloClient,
+  ApolloClientOptions,
+  OperationVariables,
+  WatchQueryOptions,
+  Observable,
+  FetchResult,
+  DocumentNode,
+  DocumentTransform,
+} from "@apollo/client";
+import { print } from "graphql";
+import { canonicalStringify } from "@apollo/client/cache";
+import { RehydrationContextValue } from "./types";
+import { registerLateInitializingQueue } from "./lateInitializingQueue";
+import {
+  ApolloBackgroundQueryTransport,
+  ApolloResultCache,
+} from "./ApolloRehydrateSymbols";
+
+const seenDocuments = new Map<string, DocumentNode>();
+
+export class NextSSRApolloClient<
+  TCacheShape
+> extends ApolloClient<TCacheShape> {
+  private rehydrationContext: Pick<
+    RehydrationContextValue,
+    "incomingBackgroundQueries"
+  > & { uninitialized?: boolean } = {
+    incomingBackgroundQueries: [],
+    uninitialized: true,
+  };
+
+  constructor(options: ApolloClientOptions<TCacheShape>) {
+    super({
+      ...options,
+      // TODO: the memoization in `documentTransform` can be removed once
+      // https://github.com/apollographql/apollo-client/pull/10968 is merged.
```

Comment:
This was just merged, I should be able to remove this once the next AC beta is released 🎉

Reply:
After upgrading to beta 3, the client-side fetch is no longer being deduped. Investigating...

Reply:
Fixed in [9a74dc6](https://github.com/apollographql/apollo-client-nextjs/pull/38/commits/9a74dc607f20aa1239cfdeee31f0e0eb157b5e43)